src/System.Runtime.Analyzers/UnitTests/SpecifyCultureInfoTests.cs

```diff
@@ -9,12 +10,480 @@ public class SpecifyCultureInfoTests : DiagnosticAnalyzerTestBase
+        {
+            VerifyCSharp(@"
+using System;
+using System.Globalization;
+
+public class CultureInfoTestClass2
+{
+    public static void Method()
+    {
+        MethodOverloadHasCultureInfoAsLastArgument(""Foo"");
+    }
+
+    public static void MethodOverloadHasCultureInfoAsLastArgument(string format)
+    {
+        MethodOverloadHasCultureInfoAsLastArgument(format, CultureInfo.CurrentCulture);
+    }
+
+    public static void MethodOverloadHasCultureInfoAsLastArgument(string format, CultureInfo provider)
+    {
+        Console.WriteLine(string.Format(provider, format));
+    }
+
+    public static void MethodOverloadHasCultureInfoAsLastArgument(CultureInfo provider, string format)
+    {
+        Console.WriteLine(string.Format(provider, format));
+    }
+}",
+            GetCSharpResultAt(9, 9, SpecifyCultureInfoAnalyzer.Rule, "CultureInfoTestClass2.MethodOverloadHasCultureInfoAsLastArgument(string)", "CultureInfoTestClass2.Method()", "CultureInfoTestClass2.MethodOverloadHasCultureInfoAsLastArgument(string, System.Globalization.CultureInfo)"));
+        }
+
+        [Fact]
+        public void CA1304_MethodOverloadHasJustCultureInfo_CSharp()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Globalization;
+
+public class CultureInfoTestClass2
+{
+    public static void Method()
+    {
+        MethodOverloadHasJustCultureInfo();
+    }
+
+    public static void MethodOverloadHasJustCultureInfo()
+    {
+        MethodOverloadHasJustCultureInfo(CultureInfo.CurrentCulture);
+    }
+
+    public static void MethodOverloadHasJustCultureInfo(CultureInfo provider)
+    {
+        Console.WriteLine(string.Format(provider, """"));
+    }
+}",
+            GetCSharpResultAt(9, 9, SpecifyCultureInfoAnalyzer.Rule, "CultureInfoTestClass2.MethodOverloadHasJustCultureInfo()", "CultureInfoTestClass2.Method()", "CultureInfoTestClass2.MethodOverloadHasJustCultureInfo(System.Globalization.CultureInfo)"));
+        }
+
+        [Fact]
+        public void CA1304_MethodOverloadCount3_CSharp()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Globalization;
+
+public class CultureInfoTestClass2
+{
+    public static void Method()
+    {
+        MethodOverloadCount3();
+    }
+
+    public static void MethodOverloadCount3()
+    {
+        MethodOverloadCount3(CultureInfo.CurrentCulture);
+    }
+
+    public static void MethodOverloadCount3(CultureInfo provider)
+    {
+        Console.WriteLine(string.Format(provider, """"));
+    }
+    public static void MethodOverloadCount3(string b)
+    {
+    }
+}",
+            GetCSharpResultAt(9, 9, SpecifyCultureInfoAnalyzer.Rule, "CultureInfoTestClass2.MethodOverloadCount3()", "CultureInfoTestClass2.Method()", "CultureInfoTestClass2.MethodOverloadCount3(System.Globalization.CultureInfo)"));
+        }
+
+        [Fact]
+        public void CA1304_MethodOverloadWithJustCulltureInfoAsExtraParameter_CSharp()
```

Comment:
How does this test differ from `CA1304_MethodOverloadHasCultureInfoAsLastArgument_CSharp`? (aside from having one more parameter). Is this test necessary?